apps/federation/lib/SyncFederationAddressBooks.php

```diff
@@ -52,6 +52,10 @@ public function syncThemAll(\Closure $callback) {
 			try {
 				$newToken = $this->syncService->syncRemoteAddressBook($url, $cardDavUser, $addressBookUrl, $sharedSecret, $syncToken, $targetBookId, $targetPrincipal, $targetBookProperties);
```

Comment:
```suggestion
				do {
					$newToken = $this->syncService->syncRemoteAddressBook($url, $cardDavUser, $addressBookUrl, $sharedSecret, $syncToken, $targetBookId, $targetPrincipal, $targetBookProperties);
				} while (str_contains($newToken, 'init_'));
```

Then it's not necessary to introduce a new method.